src/main/java/net/minecraftforge/event/TickEvent.java

```diff
@@ -45,9 +48,18 @@ public TickEvent(Type type, LogicalSide side, Phase phase)
     }
 
     public static class ServerTickEvent extends TickEvent {
-        public ServerTickEvent(Phase phase)
+
+        private final BooleanSupplier haveTime;
+
+        public ServerTickEvent(Phase phase, BooleanSupplier haveTime)
         {
             super(Type.SERVER, LogicalSide.SERVER, phase);
+            this.haveTime = haveTime;
+        }
+
+        public boolean haveTime()
```

Comment:
I think this and its sister method in `WorldTickEvent` will benefit from having a small explanatory javadoc about what this is and possibly what it is useful for.